addons/zebra-striping/userscript.js

```diff
@@ -1,69 +1,82 @@
 import { updateAllBlocks } from "../../libraries/common/cs/update-all-blocks.js";
 
 export default async function ({ addon, msg, console }) {
-  const ScratchBlocks = await addon.tab.traps.getBlockly();
+  await addon.tab.loadScript("/libraries/thirdparty/cs/tinycolor-min.js");
 
+  const ScratchBlocks = await addon.tab.traps.getBlockly();
   const originalRender = ScratchBlocks.BlockSvg.prototype.render;
   ScratchBlocks.BlockSvg.prototype.render = function (opt_bubble) {
+    const shade = addon.settings.get("shade");
+    const intensity = addon.settings.get("intensity");
+    const amount = ((shade === "lighter" ? 1 : -1) * intensity) / 2;
+
     // Any changes that affect block striping should bubble to the top block of the script.
     // The top block of the script is responsible for striping all of its children.
     // This way stripes are computed exactly once.
-    if (!this.isInFlyout && !this.isShadow() && this.getParent() === null) {
-      const stripeState = new Map();
-      // Conveniently getDescendants() returns blocks in an order such that each block's
-      // parent will always come before that block (except the first block which has no
-      // parent).
+    if (
+      // addon is enabled
+      !addon.self.disabled &&
+      // not in the flyout
+      !this.isInFlyout &&
+      // not a shadow
+      !this.isShadow() &&
+      // not an insertion marker
+      !this.isInsertionMarker() &&
+      // does not have parent
+      this.getParent() === null
+    ) {
+      // getDescendants() returns blocks in an order such that each block's parent
+      // will always come before that block (except the first block which has no parent).
       for (const block of this.getDescendants()) {
         const parent = block.getSurroundParent();
 
-        let isStriped = false;
-        if (parent) {
-          if (block.isShadow()) {
-            isStriped = !!stripeState.get(parent);
-          } else if (parent.getColour() === block.getColour()) {
-            isStriped = !stripeState.get(parent);
-          }
+        const shouldStripe =
+          // not a insertion marker
+          !block.isInsertionMarker() &&
+          // has a parent
+          parent &&
+          // shadow blocks should be striped if it's parent is striped. otherwise, it should be opposite of parent.
+          (block.isShadow() ? parent.saStriped : !parent.saStriped) &&
+          // parent and child are same color (even shadows match block color)
+          // we dont check category because other addons can make blocks the same color.
+          // if the block was stripped we need to look at it's original color in order to get it's real color.
+          parent.getColour() === (block.saOrginalColour ? block.saOrginalColour[0] : block.getColour());
```

Comment:
This still doesn't work for shadow blocks - I thought shadow dropdowns would have the same color as the parent block, but that doesn't seem to be the case. Comparing the secondary color if `block` is a shadow works though, but the parent's `saOriginalColor` needs to be checked too. The correct result looks like this:
![image](https://github.com/user-attachments/assets/99d00d13-9974-4733-805f-4f6ca7147dbb)

The current code makes it look like this instead:
![image](https://github.com/user-attachments/assets/daa784e7-6b1e-4e24-94f9-83471b95d0f3)

Reply:
@mxmou maybe i can have shadow blocks check if there parent is the same, but all other regular blocks would check primary color. that might make the most sense.

Reply:
> maybe i can have shadow blocks check if there parent is the same

If you meant "is the same category", that wouldn't work because the category of many shadow blocks is unfortunately null. Maybe you could check if the shadow block contains a dropdown field?